codex-rs/cloud-tasks-client/src/http.rs

```diff
@@ -0,0 +1,737 @@
+use crate::ApplyOutcome;
+use crate::ApplyStatus;
+use crate::AttemptStatus;
+use crate::CloudBackend;
+use crate::CloudTaskError;
+use crate::DiffSummary;
+use crate::Result;
+use crate::TaskId;
+use crate::TaskStatus;
+use crate::TaskSummary;
+use crate::TurnAttempt;
+use crate::api::TaskText;
+use chrono::DateTime;
+use chrono::Utc;
+
+use serde_json::Value;
+use std::cmp::Ordering;
+use std::collections::HashMap;
+
+use codex_backend_client as backend;
+use codex_backend_client::CodeTaskDetailsResponseExt;
+
+#[derive(Clone)]
+pub struct HttpClient {
+    pub base_url: String,
+    backend: backend::Client,
+}
+
+impl HttpClient {
```

Comment:
Structurally, it doesn't feel right that all of the logic for every API is in one giant impl. Can this be structured such that each API can have separate business logic?